tools/dotcms-cli/cli/src/main/java/com/dotcms/cli/command/files/FilesPush.java

```diff
@@ -76,127 +53,134 @@ public class FilesPush extends AbstractFilesCommand implements Callable<Integer>
-                    }
-
-                    if (dryRun) {
-                        TreePrinter.getInstance().formatByStatus(
-                                sb,
-                                AssetsUtils.statusToBoolean(localPathStructure.status()),
-                                List.of(localPathStructure.language()),
-                                treeNode,
-                                false,
-                                true,
-                                localPathStructure.languageExists());
-                    }
-
-                    output.info(sb.toString());
-
-                    // ---
-                    // Pushing the tree
-                    if (!dryRun) {
-                        pushService.processTreeNodes(output, workspace.getAbsolutePath(),
-                                localPathStructure, treeNode, treeNodePushInfo, failFast, retryAttempts);
-                    }
+        CompletableFuture<List<Triple<List<Exception>, AssetsUtils.LocalPathStructure, TreeNode>>>
+                folderTraversalFuture = CompletableFuture.supplyAsync(
+                () -> {
+                    // Service to handle the traversal of the folder
+                    return pushService.traverseLocalFolders(output, workspace, pushMixin.path,
+                            filesPushMixin.removeAssets, filesPushMixin.removeFolders,
+                            true, true);
+                });
+
+        // ConsoleLoadingAnimation instance to handle the waiting "animation"
+        ConsoleLoadingAnimation consoleLoadingAnimation = new ConsoleLoadingAnimation(
+                output,
+                folderTraversalFuture
+        );
+
+        CompletableFuture<Void> animationFuture = CompletableFuture.runAsync(
+                consoleLoadingAnimation
+        );
+
+        // Waits for the completion of both the folder traversal and console loading animation tasks.
+        // This line blocks the current thread until both CompletableFuture instances
+        // (folderTraversalFuture and animationFuture) have completed.
+        CompletableFuture.allOf(folderTraversalFuture, animationFuture).join();
+        final var result = folderTraversalFuture.get();
+
+        if (result == null) {
+            output.error(String.format(
+                    "Error occurred while pushing folder info: [%s].", pushMixin.path));
+            return CommandLine.ExitCode.SOFTWARE;
+        }
 
+        // Let's try to print these tree with some order
+        result.sort((o1, o2) -> {
+            var left = o1.getMiddle();
+            var right = o2.getMiddle();
+            return left.filePath().compareTo(right.filePath());
+        });
+
+        var count = 0;
+
+        for (var treeNodeData : result) {
+
+            var localPathStructure = treeNodeData.getMiddle();
+            var treeNode = treeNodeData.getRight();
+
+            StringBuilder sb = new StringBuilder();
```

Comment:
sb is too generic, rename to messageSomething or something else

Reply:
Done